packages/http/test/routes.test.ts

```diff
@@ -508,3 +509,128 @@ describe("http: routes", () => {
+      const param = await getParameter(`@route("/bar{?foo}") op foo(foo: string): void;`, "foo");
+      strictEqual(param.type, "query");
+    });
+
+    it("extract simple query continuation parameter", async () => {
+      const param = await getParameter(
+        `@route("/bar?fixed=yes{&foo}") op foo(foo: string): void;`,
+        "foo"
+      );
+      strictEqual(param.type, "query");
+    });
+  });
+
+  describe("build uriTemplate from parameter", () => {
+    it.each([
+      ["@path one: string", "/foo/{one}"],
+      ["@path(#{allowReserved: true}) one: string", "/foo/{+one}"],
+      ["@path(#{explode: true}) one: string", "/foo/{one*}"],
+      [`@path(#{style: "matrix"}) one: string`, "/foo/{;one}"],
+      [`@path(#{style: "label"}) one: string`, "/foo/{.one}"],
+      [`@path(#{style: "fragment"}) one: string`, "/foo/{#one}"],
+      [`@path(#{style: "path"}) one: string`, "/foo/{/one}"],
+      ["@path(#{allowReserved: true, explode: true}) one: string", "/foo/{+one*}"],
+      ["@query one: string", "/foo{?one}"],
+    ])("%s -> %s", async (param, expectedUri) => {
+      const op = await getOp(`@route("/foo") op foo(${param}): void;`);
+      expect(op.uriTemplate).toEqual(expectedUri);
+    });
+  });
+
+  it("emit diagnostic when annotating a path parameter with @query", async () => {
+    const diagnostics = await diagnoseOperations(
+      `@route("/bar/{foo}") op foo(@query foo: string): void;`
+    );
+    expectDiagnostics(diagnostics, {
+      code: "@typespec/http/incompatible-uri-param",
+      message: "Parameter 'foo' is defined in the uri as a path but is annotated as a query.",
+    });
+  });
+
+  it("emit diagnostic when annotating a query parameter with @path", async () => {
+    const diagnostics = await diagnoseOperations(
+      `@route("/bar/{?foo}") op foo(@path foo: string): void;`
+    );
+    expectDiagnostics(diagnostics, {
+      code: "@typespec/http/incompatible-uri-param",
+      message: "Parameter 'foo' is defined in the uri as a query but is annotated as a path.",
+    });
+  });
+
+  it("emit diagnostic when annotating a query continuation parameter with @path", async () => {
+    const diagnostics = await diagnoseOperations(
+      `@route("/bar/?bar=def{&foo}") op foo(@path foo: string): void;`
+    );
+    expectDiagnostics(diagnostics, {
+      code: "@typespec/http/incompatible-uri-param",
+      message: "Parameter 'foo' is defined in the uri as a query but is annotated as a path.",
+    });
+  });
+
+  describe("emit diagnostic if using any of the path options when parameter is already defined in the uri template", () => {
+    it.each([
+      "#{ allowReserved: true }",
+      "#{ explode: true }",
+      `#{ style: "label" }`,
+      `#{ style: "matrix" }`,
+      `#{ style: "fragment" }`,
+      `#{ style: "path" }`,
+    ])("%s", async (options) => {
+      const diagnostics = await diagnoseOperations(
+        `@route("/bar/{foo}") op foo(@path(${options}) foo: string): void;`
```

Comment:
It depends, if you define the parameter in the uri template you MUST define the attributes via the template so if you redefine in `@path` it will report a error. But in case of autoRoute where you don't explicitly add param in the uri you can also use those decorator attribute to achieve the same result